bitcoind-regtest/src/Bitcoin/Core/Regtest/Generator.hs

```diff
@@ -52,129 +89,242 @@ generateWithTransactions ::
     -- | Mean fee rate at height
     (BlockHeight -> Word64) ->
     IO ()
-generateWithTransactions mgr nodeHandle blockInterval getExternalAddress getMeanFeeRate =
-    either throwIO pure =<< runBitcoind mgr nodeHandle generator
+generateWithTransactions mgr nodeHandle blockInterval getExternalAddress getMeanFeeRate = do
+    run initialize
+
+    (async >=> link) . run . RPC.withWallet miningWallet . waitRepeat 1 $ do
+        balance <- getBalance
+        when (balance >= faucetPayment) $
+            liftIO getExternalAddress >>= mapM_ fundAddress
+
+    run . flip evalStateT mempty . waitRepeat blockInterval $ do
+        mapStateT (RPC.withWallet floodingWallet) $ txFlood getMeanFeeRate
+        lift . generatorGenerate =<< getSeriesCount
   where
-    generator = flip evalStateT emptyState $ do
-        initialize
-        forever $ do
-            sendTransactions
-            generatorGenerate
-            liftIO $ threadDelay (blockInterval * 1_000_000)
+    run = runBitcoind mgr nodeHandle >=> either throwIO pure
+
+    miningWallet = "bitcoin-regtest-generator.mine"
+    floodingWallet = "bitcoin-regtest-generator.flood"
+    password = "password"
```

Comment:
What's going on here?

Reply:
I found it to be easier to express this logic if there are two wallets, one dedicated to funding the flood transactions and one for mining and servicing faucet requests.  The flooding wallet only ever has the right sized outputs in it, whereas the mining wallet has a hodge-podge.

Reply:
What's the `password = "password"` for? That seems weird.

Reply:
Passing an empty password creates an unencrypted wallet.  So I refactored the generator code and some of the test code to use an unencrypted wallet.